Userland/Libraries/LibC/scanf.cpp

```diff
@@ -243,127 +246,126 @@ struct ReadElementConcrete<float, ApT, kind> {
+                return ReadElementConcrete<T, long, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
             if constexpr (IsSame<T, unsigned>)
-                return ReadElementConcrete<T, unsigned long, kind> {}(input_lexer, ap, suppress_assignment);
+                return ReadElementConcrete<T, unsigned long, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
             if constexpr (IsSame<T, float>)
-                return ReadElementConcrete<int, double, kind> {}(input_lexer, ap, suppress_assignment);
+                return ReadElementConcrete<float, double, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
             return false;
         case LengthModifier::LongLong:
             if constexpr (IsSame<T, int>)
-                return ReadElementConcrete<long long, long long, kind> {}(input_lexer, ap, suppress_assignment);
+                return ReadElementConcrete<long long, long long, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
             if constexpr (IsSame<T, unsigned>)
-                return ReadElementConcrete<unsigned long long, unsigned long long, kind> {}(input_lexer, ap, suppress_assignment);
-            if constexpr (IsSame<T, float>)
-                return ReadElementConcrete<long long, double, kind> {}(input_lexer, ap, suppress_assignment);
+                return ReadElementConcrete<unsigned long long, unsigned long long, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
             return false;
         case LengthModifier::IntMax:
-            return ReadElementConcrete<T, intmax_t, kind> {}(input_lexer, ap, suppress_assignment);
+            return ReadElementConcrete<T, intmax_t, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
         case LengthModifier::Size:
-            return ReadElementConcrete<T, size_t, kind> {}(input_lexer, ap, suppress_assignment);
+            return ReadElementConcrete<T, size_t, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
         case LengthModifier::PtrDiff:
-            return ReadElementConcrete<T, ptrdiff_t, kind> {}(input_lexer, ap, suppress_assignment);
+            return ReadElementConcrete<T, ptrdiff_t, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
         case LengthModifier::LongDouble:
-            return ReadElementConcrete<T, long double, kind> {}(input_lexer, ap, suppress_assignment);
+            return ReadElementConcrete<T, long double, kind> {}(input_lexer, ap, width_specifier, suppress_assignment);
         }
     }
 };
 
 template<>
 struct ReadElement<char*, ReadKind::Normal> {
-    ReadElement(StringView scan_set = {}, bool invert = false)
-        : scan_set(scan_set.is_null() ? " \t\n\f\r"sv : scan_set)
-        , invert(scan_set.is_null() ? true : invert)
+    ReadElement(size_t width_specifier, StringView scan_set = {}, bool invert = true)
```

Comment:
Did the default of `invert` change intentionally?

Reply:
Yes. because `true` is the actual default of `invert` in the old code.

`invert` is true when `scan_set` is the default. 
https://github.com/SerenityOS/serenity/blob/5ed7cd6e320ce5bd3882adc8b65ea59131fca01b/Userland/Libraries/LibC/scanf.cpp#L290

When we specify `scan_set`, we specify `invert` as well.
https://github.com/SerenityOS/serenity/blob/5ed7cd6e320ce5bd3882adc8b65ea59131fca01b/Userland/Libraries/LibC/scanf.cpp#L589